app/jobs/course/user_deletion_job.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+class Course::UserDeletionJob < ApplicationJob
+  protected
+
+  def perform(course, course_user, current_user)
+    @course = course
+    @course_user = course_user
+    @current_user = current_user
+
+    ActsAsTenant.without_tenant do
+      success = delete_course_user
+
+      handle_deletion_failure unless success
+    end
+  end
+
+  def delete_course_user
+    @course_user.class.transaction do
+      raise ActiveRecord::Rollback unless @course_user.destroy
+
+      true
+    end
+  end
```

Comment:
Since only 1 operation (wrapped in `@course_user.destroy`, don't need to use transaction block.

then you can simply do:

`handle_deletion_failure unless @course_user.destroy`

Reply:
done. I just checked and the operation `destroy` itself is a transaction; if it failed to destroy, all the child objects will be reinstated back.

I have also simplified the code inside this file not to introduce many private functions as well